sklearn/kernel_approximation.py

```diff
@@ -676,27 +686,40 @@ def fit(self, X, y=None):
             Returns the transformer.
         """
         self._validate_params()
-
         X = self._validate_data(X, accept_sparse="csr")
         check_non_negative(X, "X in AdditiveChi2Sampler.fit")
 
+        # TODO(1.5): remove the setting of _sample_interval from fit
         if self.sample_interval is None:
-            # See reference, figure 2 c)
+            # See figure 2 c) of "Efficient additive kernels via explicit feature maps"
+            # <http://www.robots.ox.ac.uk/~vedaldi/assets/pubs/vedaldi11efficient.pdf>
+            # A. Vedaldi and A. Zisserman, Pattern Analysis and Machine Intelligence,
+            # 2011
             if self.sample_steps == 1:
-                self.sample_interval_ = 0.8
+                self._sample_interval = 0.8
```

Comment:
I see that we have lines that are not covered. Indeed, I think that we can cover them by calling `fit_transform` and `transform` for the same set of parameters `sample_steps`.